boards/x86/up_squared/doc/up_squared.rst

```diff
@@ -0,0 +1,238 @@
+addresses and sizes.
+
+Programming and Debugging
+*************************
+
+Use the following procedures for booting an image on a Galileo board.
+
+* `Creating a GRUB2 Boot Loader Image from a Linux Host`_
+
+* `Preparing the Boot Device`_
+
+* `Booting the UP Squared Board`_
+
+Creating a GRUB2 Boot Loader Image from a Linux Host
+====================================================
+
+If you are having problems running an application using the preinstalled
+copy of GRUB, follow these steps to test on supported boards using a custom GRUB.
+
+#. Install the requirements to build GRUB on your host machine.
+
+   On Ubuntu, type:
+
+   .. code-block:: console
+
+      $ sudo apt-get install bison autoconf libopts25-dev flex automake
+
+   On Fedora, type:
+
+   .. code-block:: console
+
+     $ sudo dnf install gnu-efi bison m4 autoconf help2man flex \
+        automake texinfo
+
+#. Clone and build the GRUB repository using the script in Zephyr tree, type:
+
+   .. code-block:: console
+
+     $ cd $ZEPHYR_BASE
+     $ ./boards/x86/common/scripts/build_grub.sh x86_64
+
+#. Find the binary at
+   :file:`$ZEPHYR_BASE/boards/x86/common/scripts/grub/bin/grub_x86_64.efi`.
+
+Preparing the Boot Device
+=========================
+
+Prepare a USB flash drive to boot the Zephyr application image on
+a UP Squared board.
+
+#. Build a Zephyr application; for instance, to build the ``hello_world``
+   application on Galileo:
+
+   .. zephyr-app-commands::
+      :zephyr-app: samples/hello_world
+      :board: up_squared
+      :goals: build
+
+   .. note::
+
+      A stripped project image file named :file:`zephyr.strip` is automatically
+      created in the build directory after the application is built. This image
+      has removed debug information from the :file:`zephyr.elf` file.
+
+#. Refer to the UP Squared Serial Console Wiki page for instructions on how to
+   connect for serial console.
+
+   `<https://wiki.up-community.org/Serial_console>`_
+
+#. Format the USB flash drive as FAT
+
+#. Create the following directories
+
+   :file:`efi`
+
+   :file:`efi/boot`
+
+   :file:`kernel`
+
+#. Copy the kernel file :file:`zephyr/zephyr.strip` to the :file:`$USB/kernel` folder.
+
+#. Copy your built version of GRUB to :file:`$USB/efi/boot/bootx64.efi`
+
+#. Create :file:`$USB/efi/boot/grub.cfg` containing the following:
+
+   .. code-block:: console
+
+      set default=0
+      set timeout=10
+
+      menuentry "Zephyr Kernel" {
+         multiboot /kernel/zephyr.strip
+      }
+
+Booting the UP Squared Board
+============================
+
+Boot the UP Squared board from the boot device using GRUB2 via USB flash drive.
+
+Steps
+-----
+
+1. Insert the prepared boot device (USB flash drive) into the UP Squared board.
+
+2. Connect the board to the host system using the serial cable and
+   configure your host system to watch for serial data.  See
+   `https://wiki.up-community.org/Serial_console`_.
```

Comment:
Replace this line with just the url:

    https://wiki.up-community.org/Serial_console